tests/Unit/ControlSystem/Test_FunctionOfTime.cpp

```diff
@@ -0,0 +1,133 @@
+    const auto& lambdas = fvt.operator()<ret_deriv_order>(t);
+
+    CHECK(lambdas.size() == ret_deriv_order + 1);
+    CHECK(approx(lambdas[1][0]) == 3.0 * square(t));
+    CHECK(approx(lambdas[1][1]) == 2.0 * t);
+
+    t += dt;
+    fvt.update(t, {6.0, 0.0});
+  }
+}
+
+SPECTRE_TEST_CASE(
+    "Unit.ControlSystem.FunctionOfTime.PiecewisePolynomial.NonConstDerivUpdate",
+    "[ControlSystem][Unit]") {
+  double t = 0.0;
+  const double dt = 0.6;
+  const double final_time = 4.0;
+  const constexpr size_t deriv_order = 2;
+
+  using value_type = std::array<DataVector, deriv_order + 1>;
+  // initally x**2, but update with non-constant 2nd deriv
+  const value_type init_func{{{0.0}, {0.0}, {2.0}}};
+  FunctionOfTime::PiecewisePolynomial<deriv_order> fvt(t, init_func);
+
+  while (t < final_time) {
+    t += dt;
+    fvt.update(t, {3.0 + t});
+  }
+  const auto& lambdas = fvt(t);
+  CHECK(approx(lambdas[0][0]) == 33.948);
+  CHECK(approx(lambdas[1][0]) == 19.56);
+  CHECK(approx(lambdas[2][0]) == 7.2);
+}
+
+// [[OutputRegex, t must be increasing from call to call.]]
+SPECTRE_TEST_CASE(
+    "Unit.ControlSystem.FunctionOfTime.PiecewisePolynomial.BadUpdateTime",
+    "[ControlSystem][Unit]") {
+  ERROR_TEST();
+  // two component system (x**3 and x**2)
+  const constexpr size_t deriv_order = 3;
+  using value_type = std::array<DataVector, deriv_order + 1>;
+  const value_type init_func{{{0.0, 0.0}, {0.0, 0.0}, {0.0, 2.0}, {6.0, 0.0}}};
+  // test non-increasing t error
+  FunctionOfTime::PiecewisePolynomial<deriv_order> fvt(0.0, init_func);
+  fvt.update(2.0, {6.0, 0.0});
+  fvt.update(1.0, {6.0, 0.0});
+}
+
+// [[OutputRegex, the number of components trying to be updated \(3\) does not
+// match the number of components \(2\) in the PiecewisePolynomial.]]
+SPECTRE_TEST_CASE(
+    "Unit.ControlSystem.FunctionOfTime.PiecewisePolynomial.BadUpdateSize",
+    "[ControlSystem][Unit]") {
+  ERROR_TEST();
+  // two component system (x**3 and x**2)
+  const constexpr size_t deriv_order = 3;
+  using value_type = std::array<DataVector, deriv_order + 1>;
+  const value_type init_func{{{0.0, 0.0}, {0.0, 0.0}, {0.0, 2.0}, {6.0, 0.0}}};
+  // test wrong update size error
+  FunctionOfTime::PiecewisePolynomial<deriv_order> fvt(0.0, init_func);
+  fvt.update(1.0, {6.0, 0.0, 0.0});
+}
+
+// [[OutputRegex, requested time 0.5 precedes earliest time 1 of times.]]
+SPECTRE_TEST_CASE(
+    "Unit.ControlSystem.FunctionOfTime.PiecewisePolynomial.TimeOutOfRange",
+    "[ControlSystem][Unit]") {
+  ERROR_TEST();
+  // two component system (x**3 and x**2)
+  const constexpr size_t deriv_order = 3;
+  const size_t n_derivs = 3;
+  using value_type = std::array<DataVector, deriv_order + 1>;
+  const value_type init_func{{{1.0, 1.0}, {3.0, 2.0}, {6.0, 2.0}, {6.0, 0.0}}};
+  FunctionOfTime::PiecewisePolynomial<deriv_order> fvt(1.0, init_func);
+  fvt.update(2.0, {6.0, 0.0});
+  // check that 1-roundoff does not throw an error
+  const auto& lambdas0 = fvt(1.0 - 5.0e-16);
```

Comment:
put this in a separate test. Error tests should test only that particular error, this may not get caught if it fails